packages/ios-client/Sources/FishjamClient/ReconnectionManager.swift

```diff
@@ -39,43 +38,32 @@ class ReconnectionManager {
     }
 
     func onDisconnected() {
-        if reconnectAttempts >= reconnectConfig.maxAttempts {
-            reconnectionStatus = .ERROR
+        guard reconnectAttempts < reconnectConfig.maxAttempts else {
+            reconnectionStatus = .error
             listener.onReconnectionRetriesLimitReached()
             return
         }
 
-        if reconnectionStatus == .WAITING {
-            return
-        }
+        guard reconnectionStatus != .reconnecting else { return }
```

Comment:
I could kind of agree that it might be easier to read. However I don't think that it's that much of an issue and for me guards are much easier to spot and by extension the whole block becomes easier to understand. `if return`s just look strange for me but that could just be bacause of my long time of working with Swift 😅